src/components/Image.js

```diff
@@ -3,13 +3,28 @@ import * as React from "react";
 import ImageZoom from "react-medium-image-zoom";
 import styled from "styled-components";
 import type { SlateNodeProps as Props } from "../types";
+import { isRetinaImage } from "../lib/imageHelpers";
 
 type State = {
   hasError?: boolean,
+  imageWidth?: number,
 };
 class Image extends React.Component<Props, State> {
   state = {
     hasError: false,
+    imageWidth: undefined,
+  };
+
+  onImgLoad = async (ev: SyntheticEvent<HTMLImageElement>) => {
+    const image = ev.target;
+    const url = this.props.node.data.get("src");
+    // Skip upload stage
+    if (url.startsWith("blob:")) return;
```

Comment:
In my testing you don't need this line at all, the browser can fetch a blob uri without issue